### ISSUE TITLE
Switch Component doesn't disable click functionality when disabled

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30944]
## Description
When using a screen reader the Switch component doesn't disable click functionality.

This issue is covered in-depth by parent issue #30840, please check there for more details.

Issue #30706 is similar to this issue and should be taken into account when creating a solution.

## React Native version:
0.63

**Comment by wertlop** [comment]
Can I take this issue?

**Comment by amarlette** [comment]
@wertlop yes you are welcome to pick up this issue. Please mention the issue number in your pull request so that it links back. Thank you!

**Comment by smash-96** [comment]
@amarlette Hi, I'd like to work on this issue if it is still available
I see that the issue is still in backlog and has no updates. Kindly let me know, thankyou!

**Comment by mdaj06** [comment]
hi @amarlette! i can have a go at this if its still open

**Comment by fabriziobertoglio1987** [comment]
I closed my pr for this reason https://github.com/facebook/react-native/pull/31199#issuecomment-830738844

**Comment by fabriziobertoglio1987** [comment]
I have a pull request ready for this issue. I will publish it soon. More information in the discussion https://github.com/fabriziobertoglio1987/react-native-notes/issues/5

**Comment by fabriziobertoglio1987** [comment]
@blavalla 
I published https://github.com/facebook/react-native/pull/33070 to fix this issue. 
The PR is in DRAFT. I will complete the PR in a couple of days. 
Thanks :pray:
